plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -1278,10 +1279,95 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     return getVisibility(decl_->getAccess());
   }
 
-  bool isWritten(const clang::VarDecl* vd_) const
+  bool isWritable(const clang::QualType& param) const
```

Comment:
Rename `param` to `type_` (it's more general) otherwise rename `isWritable` to `isFuncParamWritable`.